--- libitl-0.7.0/Makefile.old	2008-06-09 15:57:37.000000000 +0800
+++ libitl-0.7.0/Makefile.in	2008-06-09 15:58:14.000000000 +0800
@@ -23,7 +23,7 @@
 	@false
 
 build/libitl.so: components
-	$(LD) build/*.o $(LDFLAGS) -shared -lm -lc -soname=$(SONAME) -o build/$(FULLNAME)
+	$(CC) build/*.o $(LDFLAGS) -shared -lm -lc -Wl,-soname,$(SONAME) -o build/$(FULLNAME)
 	(cd build/ && ln -sf $(FULLNAME) $(SONAME))
 	(cd build/ && ln -sf $(FULLNAME) libitl.so)
 